api/stats/index.js

```diff
@@ -3,14 +3,79 @@
 let request = require("leo-auth");
 let stats = require("../../lib/stats.js");
 require("moment-round");
+var zlib = require("zlib");
+
+let compressionThreshold = 100000; // 100k
 
 exports.handler = require("leo-sdk/wrappers/resource")(async (event, context, callback) => {
     await request.authorize(event, {
         lrn: 'lrn:leo:botmon:::',
         action: "stats",
         botmon: {}
     });
+
+    function findBigStrings(obj, prefix = "") {
+        if (!obj) return;
+        let objType = typeof obj;
+        if (objType === 'object') {
+            if (Array.isArray(obj)) {
+                obj.forEach((value, index) => {
+                    let localPrefix = prefix === '' ? `[${index}]` : prefix + `[${index}]`;
+                    findBigStrings(value, localPrefix);
+                });
+            }
+            else {
+                for (const key of Object.keys(obj)) {
+                    let localPrefix = prefix === '' ? key : prefix + "." + key;
+                    findBigStrings(obj[key], localPrefix);
+                }
+            }
+        } else if (objType === 'string') {
+            if (obj.length > 1024) {
+                console.log(`string stored at '${prefix}' is larger than 1024 bytes (length=${obj.length}): '${obj}'`);
+            }
+        }
+    }
+
     stats(event, (err, data) => {
-        callback(err, (data || {}).stats)
+        let stats = (data || {}).stats;
+        if (stats) {
+            let responseBody = JSON.stringify(stats);
+            console.log(`response body length = ${responseBody.length}`);
+            if (responseBody.length > 6000000) {
+                findBigStrings(stats);
+            }
+            let isBase64Encoded = false;
```

Comment:
@jgrantr is this still needed if we have MinimumCompressionSize set?

Reply:
@czirker If we are compressing, probably not, unless we end up with something too big for even compressing, and then we may want to be able to track it down.

Reply:
We will leave it in for now